scripts/install.sh

```diff
@@ -0,0 +1,233 @@
+    exit 1
+fi
+
+log INFO "Distro Information as $DISTRIB_ID"
+
+# set distribution specific vars
+PACKAGER=
+SYSTEMD_PATH=/lib/systemd/system
+if [ "$DISTRIB_ID" == "ubuntu" ]; then
+    PACKAGER=apt
+    elif [ "$DISTRIB_ID" == "debian" ]; then
+    PACKAGER=apt
+    elif [[ $DISTRIB_ID == centos* ]] || [ "$DISTRIB_ID" == "rhel" ]; then
+    PACKAGER=yum
+    elif [[ "$DISTRIB_ID" == "darwin"* ]]; then
+    PACKAGER=brew
+    elif [[ "$DISTRIB_ID" == "win32" ]] || [[ "$DISTRIB_ID" == "msys" ]]; then
+    PACKAGER=choco
+else
+    PACKAGER=zypper
+    SYSTEMD_PATH=/usr/lib/systemd/system
+fi
+if [ "$PACKAGER" == "apt" ]; then
+    export DEBIAN_FRONTEND=noninteractive
+fi
+
+# Check JQ Processor and download if not present
+check_jq_processor_present() {
+    check_if_exists jq
+    if [[ $result != 0 ]]; then
+        install_tool jq
+    fi
+    jqversion=$(jq --version)
+    log INFO "Locally installed JQ Processor version is $jqversion"
+}
+
+check_kubectl_present() {
+    check_if_exists kubectl
+    if [[ $result != 0 ]]; then
+        install_tool kubectl
+    fi
+    kubectlversion=$(kubectl version --client=true -o json | jq ".clientVersion.gitVersion")
+    log INFO "Locally installed kubectl version is $kubectlversion"
+}
+
+check_dotnet_runtime_present() {
+    check_if_exists dotnet
+    # if dotnet doesn't exist install it
+    if [[ $result != 0 ]]; then
+        install_tool dotnet
+        return;
+    fi
+    #if dotnet exists, check the version required for b2k and install it.
+    dotnetruntimes=$(dotnet --list-runtimes)
+    if [[ -z "${dotnetruntimes}" || ! "${dotnetruntimes}" =~ '3.1'* ]]; then
+        install_tool dotnet
+    else 
+        log INFO "dotnet version is $(dotnet --version)"
+    fi
+}
+
+install_tool() {
+    log INFO "installing $1.."
+
+    case $1 in 
+
+        kubectl)
+            install_with_sudo kubectl
+            ;;
+        dotnet)
+            if [[ $OSTYPE == "darwin"* ]]; then
+                $PACKAGER tap isen-ng/dotnet-sdk-versions
+                install_with_sudo dotnet-sdk3-1-400 --cask
+            else 
+                install_with_sudo aspnetcore-runtime-3.1
+            fi
+            ;;
+        jq)
+            install_with_sudo jq
+            ;;
+        *)
+            log INFO "Unknown option for install $1"
+            exit 1
+            ;;
+    esac
+}
+
+install_with_sudo() {
+    if [[ $OSTYPE == "linux"* ]]; then
+        sudo $PACKAGER install $1 -y
+    else
+        $PACKAGER install $2 $1
+    fi
+}
+
+check_if_exists() {
+    log INFO "checking if $1 exists"
+    if ! [[ -x "$(command -v $1)" ]]; then
+        log INFO "Error: $1 is not installed." >&2
+        result=1
+    else 
+        result=0
+    fi
+}
+
+# Download bridge stable version, this can be done via following command curl -LO $(curl -L -s https://aka.ms/bridge-lks | jq -r '.linux.url')
+download_bridge_stable_version() {
+    log INFO "Starting B2K Download"
```

Comment:
Why are we providing the binaries for .NET runtime and kubectl? Wouldn't it be better to get those dependencies from another official source versus hosting the binaries ourselves at the risk of not providing the latest patches?

Reply:
@sabbour valid question, this [url](https://aka.ms/bridge-lks-v2) provides all three binaries bridge, dotnet, kubectl but we do download only bridge. Kubectl and dotnet are downloaded via each OS package manager ex: for mac os it will be brew install dotnet. you can refer lines 86 to 116. thank you